### PR TITLE
AzureMonitor: Convert Logs to React

### LINKED ISSUE #30794: AzureMonitor: React Migration
This is an epic issue to collect and track the migration of the Azure Monitor datasource plugin to React, and improvements to the query editors, in a rough logical order

 - [x] Port Metrics to React
    - [x] Rewrite UI in React https://github.com/grafana/grafana/issues/31108 PR: https://github.com/grafana/grafana/pull/30783
    - [x] Error handling https://github.com/grafana/grafana/issues/31109
    - [x] Make "empty" query values actually nullable https://github.com/grafana/grafana/issues/31086
 - [x] Port Logs to React
 - [x] Loading indicators on dropdowns https://github.com/grafana/grafana/issues/31083
 - [x] Use Monaco for KQL editor PR https://github.com/grafana/grafana/pull/33528
 - [x] Deprecate Application Insights and Insights Analytics https://github.com/grafana/grafana/issues/30184
     - [x] Create read-only view for App Insights https://github.com/grafana/grafana/issues/32789
     - [x] Create read-only view for Insights Analytics https://github.com/grafana/grafana/issues/32790
 - [x] Basic UI-less Resource Picker for Logs https://github.com/grafana/grafana/issues/33700
 - [x] Resource Picker UI https://github.com/grafana/grafana/issues/33704
 - [x] Logs Resource-centric queries https://github.com/grafana/grafana/issues/33705
 - [x] Switch QueryEditor entry point to React (remove Angular) https://github.com/grafana/grafana/issues/33706
 - [x] Resource Picker for Metrics https://github.com/grafana/grafana/issues/44439
 - [ ] Use `VisualQueryBuilder` from ADX for Logs queries
     - [ ] Consider: performance of metadata queries 
     - [x] Bug: whole "where" line not removed https://github.com/grafana/azure-data-explorer-datasource/issues/121
 - [ ] Replace variable macros with query builder
 - [ ] Documentation

### PR DESCRIPTION
**Pull request by joshhunt**
Using workspace queries at the moment - moving Logs over the react now, and will do resource-centric queries later :)

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/QueryEditor.tsx [hunk 1]
```diff
@@ -25,22 +26,22 @@ const QueryEditor: React.FC<BaseQueryEditorProps> = ({ query, datasource, onChan
...
-      <VerticalGroup>
-        <EditorForQueryType
-          subscriptionId={subscriptionId}
-          query={query}
-          datasource={datasource}
-          onChange={onChange}
-          variableOptionGroup={variableOptionGroup}
-          setError={setError}
-        />
+      {/* <VerticalGroup> */}
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/QueryEditor.tsx:29]: Remove or uncomment

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/QueryEditor.tsx [hunk 2]
```diff
@@ -25,22 +26,22 @@ const QueryEditor: React.FC<BaseQueryEditorProps> = ({ query, datasource, onChan
...
-            {errorMessage}
-          </Alert>
-        )}
-      </VerticalGroup>
+      {errorMessage && (
+        <Alert severity="error" title="An error occurred while requesting metadata from Azure Monitor">
+          {errorMessage}
+        </Alert>
+      )}
+      {/* </VerticalGroup> */}
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/QueryEditor.tsx:44]: same here

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    [onQueryChange, query]
+  );
+
+  return (
+    <CodeEditor
+      value={query.azureLogAnalytics.query}
+      language="kql"
+      height={200}
+      width="100%"
+      showMiniMap={false}
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx:25]: Do you need to set this prop?

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        },
+      });
+    },
+    [onQueryChange, query]
+  );
+
+  return (
+    <CodeEditor
+      value={query.azureLogAnalytics.query}
+      language="kql"
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx:22]: Is there a language definition for KQL? I can't seem to find it [here](https://github.com/microsoft/monaco-languages/tree/main/src).

**Review by sunker — COMMENTED** [review]
I don't get any syntax highlighting. I assume that's because there's no language definition available for KQL. I wonder if that can be extracted from [here](https://github.com/Azure/monaco-kusto) in some way? 

Also I'm not getting any suggestions (apart from those that come from the existing query). We should at least be able to provide [these](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/grafana-azure-monitor-datasource/editor/kusto/kusto.ts) suggestions to Monaco right?

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx (hunk 3 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx:25]: Need? No. Want? yes!

The minimap seems distracting and offers no benefit for a small code editor like this. IMHO it should default to off for all `<CodeEditor/>`s, but that's not for here :)

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx (hunk 4 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx:22]: no not yet. i'll remove this for now :)

Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx:22]: oh. language is mandatory. I'll leave it there, it just won't do anything yet.

**Review by sunker — APPROVED** [review]
LGTM

**Pull request merged**